packages/delegation-process/test/approveDelegation.test.ts

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable functional/no-let */
+import {
+  decodeProtobufPayload,
+  getMockDelegationProducer,
+  getMockTenant,
+} from "pagopa-interop-commons-test/index.js";
+import { describe, expect, it, vi } from "vitest";
+import {
+  DelegationApprovedV2,
+  DelegationId,
+  toDelegationV2,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { delegationState } from "pagopa-interop-models";
+import {
+  delegationNotFound,
+  operationRestrictedToDelegate,
+  incorrectState,
+} from "../src/model/domain/errors.js";
+import {
+  addOneDelegation,
+  delegationProducerService,
+  readLastDelegationEvent,
+} from "./utils.js";
+
+describe("approve delegation", () => {
+  it("should approve delegation in the happy path", async () => {
+    const currentExecutionTime = new Date();
+    vi.useFakeTimers();
+    vi.setSystemTime(currentExecutionTime);
+
+    const delegate = getMockTenant();
+    const delegation = getMockDelegationProducer({
+      state: "WaitingForApproval",
+      delegateId: delegate.id,
+    });
+    await addOneDelegation(delegation);
+
+    await delegationProducerService.approveProducerDelegation(
+      delegate.id,
+      delegation.id,
+      "9999"
+    );
+
+    const event = await readLastDelegationEvent(delegation.id);
```

Comment:
IMHO: to cover the correct use case, first of all you should create an existent event in event store, to simulate the already created delegation `"DelgationSubmitted"` event. So, like in the real usages, in the second time you read event on same stream id (delegation id) that was already created.

Reply:
I'm not getting this, since we are not testing the read model writer here I cannot validate the version is increased so what is the value of checking the first is already there? I mean, If I do this check it means I'm verifying the test utilities are working fine, am I wrong?

Reply:
Commit [203083c](https://github.com/pagopa/interop-be-monorepo/pull/1103/commits/203083c4a730756bcdffb9fc064e1d235d0a6a3d) resolve this comment